sdk/core/Azure.Core/src/Pipeline/HttpPipelineBuilder.cs

```diff
@@ -105,6 +105,12 @@ internal static TelemetryPolicy CreateTelemetryPolicy(ClientOptions options)
                 assemblyName = assemblyName.Substring(PackagePrefix.Length);
             }
 
+            int hashSeparator = version.IndexOf('+');
```

Comment:
Do we know that '+' won't ever be found elsewhere in a UA header?

Reply:
This only checks the assembly informational version that only uses `+` to prefix the commit sha.